implementation/src/main/java/io/smallrye/config/SmallRyeConfigProviderResolver.java

```diff
@@ -63,20 +66,22 @@ public ConfigBuilder getBuilder() {
 
     @Override
     public void registerConfig(Config config, ClassLoader classLoader) {
-        configsForClassLoader.put(classLoader, config);
+        synchronized (configsForClassLoader) {
```

Comment:
Noob sync question here. Can you sync on a variable that's already synchronized on?

Reply:
Noob developer here :)

You are right, I need to synchronise on the instance, not the collection itself. thanks for the catch.